src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -25,7 +25,7 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
         {
             JsonTypeInfo jsonTypeInfo = state.Current.JsonTypeInfo;
 
-            if (jsonTypeInfo.CreateObject != null)
+            if (jsonTypeInfo.CreateObject != null || state.Current.IsPopulating)
             {
                 // Contract customization: fall back to default object converter if user has set a default constructor delegate.
```

Comment:
Does this comment need updating now that you've updated the logic?

Reply:
commend updated in the next iteration